Start of evaluator functions

hdemusg opened the pull request:
## Description
Rewriting the evaluator functions to be compatible with the new gameplay structure. 

Ready for review:
12 Evaluator Function Files
Ball: DONE
Chip: DONE
Defensive Position: 5/7 functions
Double Touch: 0/10 functions (skipping for now since not needed per Hussain)
Field: DONE
Linear Classification: DONE
Opponent: 2/6 functions
Pass: 0/1 function
Pass Positioning: 0/2 functions
Path: 1/3 functions
Shoot: 0/2 functions
Touchpass Position: 3/4 functions

WIP:
Everything else, with emphasis on functions that do not require KickEvaluator or WindowEvaluator from the old rc code.

hdemusg committed:
started new eval functions

hdemusg committed:
merge

hdemusg committed:
first eval complete

HussainGynai reviewed (changes requested):
You made your PR from ros2eval to staging, it should be from ros2eval to ros2. That's why its says you have 1000+ files changed.

hdemusg committed:
Merge remote-tracking branch 'origin/ros2' into ros2eval

hdemusg committed:
3/30 updates

hdemusg committed:
Create pass_positioning.py

hdemusg committed:
Create path.py

hdemusg committed:
Create shoot.py

hdemusg committed:
4/10 updates to Touchpass

hdemusg committed:
moved pass definition to rc

hdemusg committed:
Update pass.py

hdemusg committed:
4/12 - updates to Pass

added math library import

hdemusg committed:
4/20 - updates to Ball

worked on all of the methods in ball.py, but none are fully completed.

hdemusg committed:
Defensive Position updates - 5/4

getting position, angle, and angle velocity of robot

hdemusg committed:
field updates - 5/5

finished space coeff at pos function!

hdemusg committed:
5/9 touchpass position changes

hdemusg committed:
5/9 changes to pass

hdemusg committed:
5/11 - ball changes

finished backward motion function

hdemusg committed:
5/12 changes to pass positioning

hdemusg committed:
creating constants.py

hdemusg committed:
Kevin's update to constants

hdemusg committed:
5/22 - changes to Ball

hdemusg committed:
Kevin's updated constants file

5/26

hdemusg committed:
Completed robot with ball functions

hdemusg committed:
more ball updates - 5/28

hdemusg committed:
finished ball

hdemusg committed:
Chip - 6/1

hdemusg committed:
removed comments from chip

hdemusg committed:
fixed goalie zone logic

hdemusg committed:
Goalie Zone line clarification

Realized that in soccer, the penalty box line counts as inside the penalty box.

hdemusg committed:
kick direction, points from rec

hdemusg committed:
create area defense zones

hdemusg committed:
6/10 - changes to Defensive Position

hdemusg committed:
more progress on estimate score

hdemusg committed:
Moved most of touchpass_position

hdemusg committed:
defensive position - 6/13 at 12:28am

hdemusg committed:
Field complete

hdemusg committed:
field updates - 6/13 at 12:52 am

hdemusg committed:
pass changes at 1:03 am

hdemusg committed:
pass positioning changes at 1:15am

hdemusg committed:
path updates - 6/13 at 1:25am

hdemusg committed:
shoot changes at 6/13 1:37am

hdemusg committed:
Finished all DefPos methods w/o kickeval

hdemusg committed:
Opponent changes
